insert_string_tpl.h

```diff
@@ -31,7 +31,7 @@
 #ifndef HASH_CALC_READ
 #  ifdef UNALIGNED_OK
 #    define HASH_CALC_READ \
-        val = *(uint32_t *)(strstart);
+        memcpy(&val, strstart, sizeof(val));
```

Comment:
why is this guarded by UNALIGNED_OK? it seems equivalent to #else case below now

Reply:
The only thing I can think of is maybe it would be better if it checked for big endian rather than unaligned ok. I have noticed it in one of my other PRs but not sure about it. Feel free to open a PR.

Reply:
Obviously on big-endian architectures the reads need to be done byte-by-byte and combined one-by-one as 32-bit variable, otherwise the CPU would trap because of misalignment. Doing byte-by-byte also makes sure little- and big-endian architectures give exactly same result when the bytes are combined as 32-bit unsigned integer.